nilearn/datasets/atlas.py

```diff
@@ -320,104 +325,274 @@ def fetch_atlas_harvard_oxford(atlas_name, data_dir=None,
         resume=resume, verbose=verbose)
-
+    # Reorder image to have positive affine diagonal
+    atlas_img = reorder_img(atlas_img)
     names = {}
     from xml.etree import ElementTree
     names[0] = 'Background'
     for label in ElementTree.parse(label_file).findall('.//label'):
-        names[int(label.get('index')) + 1] = label.text
-    names = list(names.values())
-
-    if not symmetric_split:
-        return Bunch(maps=atlas_img, labels=names)
-
-    if atlas_name in ("cort-prob-1mm", "cort-prob-2mm",
-                      "sub-prob-1mm", "sub-prob-2mm"):
-        raise ValueError("Region splitting not supported for probabilistic "
-                         "atlases")
-
-    atlas_img = check_niimg(atlas_img)
-    if lateralized:
-        return Bunch(maps=atlas_img, labels=names)
-
-    atlas = get_data(atlas_img)
-
-    labels = np.unique(atlas)
+        new_idx = int(label.get('index')) + 1
+        if new_idx in names:
+            raise IndexError(
+                f"Duplicate index {new_idx} for labels "
+                f"'{names[new_idx]}', and '{label.text}'")
+        names[new_idx] = label.text
+    names = [item[1] for item in sorted(names.items())]
+    return atlas_img, names, is_lateralized
+
+
+def _merge_probabilistic_maps_juelich(atlas_data, names):
+    """Helper function for fetch_atlas_juelich.
+    This function handles probabilistic juelich atlases
+    when symmetric_split=False. In this situation, we need
+    to merge labels and maps corresponding to left and right
+    regions.
+    """
+    new_names = np.unique([re.sub(r" (L|R)$", "", name) for name in names])
+    new_name_to_idx = {k: v - 1 for v, k in enumerate(new_names)}
+    new_atlas_data = np.zeros((*atlas_data.shape[:3],
+                               len(new_names) - 1))
+    for i, name in enumerate(names):
+        if name != "Background":
+            new_name = re.sub(r" (L|R)$", "", name)
+            new_atlas_data[..., new_name_to_idx[new_name]] += (
+                atlas_data[..., i - 1])
+    return new_atlas_data, new_names
+
+
+def _merge_labels_juelich(atlas_data, names):
+    """Helper function for fetch_atlas_juelich.
+    This function handles 3D atlases when symmetric_split=False.
+    In this case, we need to merge the labels corresponding to
+    left and right regions.
+    """
+    new_names = np.unique([re.sub(r" (L|R)$", "", name) for name in names])
+    new_names_dict = {k: v for v, k in enumerate(new_names)}
+    new_atlas_data = atlas_data.copy()
+    for label, name in enumerate(names):
+        new_name = re.sub(r" (L|R)$", "", name)
+        new_atlas_data[atlas_data == label] = new_names_dict[new_name]
+    return new_atlas_data, new_names
+
+
+def _compute_symmetric_split(source, atlas_niimg, names):
+    """Helper function for both fetch_atlas_juelich and
+    fetch_atlas_harvard_oxford.
+    This function handles 3D atlases when symmetric_split=True.
+    """
+    atlas_data = get_data(atlas_niimg)
```

Comment:
maybe assert atlas_niimg.affine[0, 0] > 0 to make it clear to someone reading this function, as the `reorder_img` happens somewhere else